src/features/content-explorer/content-explorer/__tests__/ContentExplorer.test.js

```diff
@@ -787,4 +804,23 @@ describe('features/content-explorer/content-explorer/ContentExplorer', () => {
             expect(instance.unselectAll).toHaveBeenCalledTimes(0);
         });
     });
+
+    describe('handleItemClick()', () => {
+        test('should update selectedItems state correctly with autoSelectedItems when new item is clicked', () => {
+            const items = [
+                { id: 'item1', name: 'name1' },
+                { id: 'item2', name: 'name2' },
+            ];
+            const autoSelectedItems = { item1: { id: 'item1', name: 'name1' } };
+            const mockEvent = { stopPropagation: () => {} };
+
+            const wrapper = renderComponent({
+                items,
+                autoSelectedItems,
+                contentExplorerMode: ContentExplorerModes.MULTI_SELECT,
+            });
+            wrapper.instance().handleItemClick({ event: mockEvent, index: 1 });
+            expect(Object.keys(wrapper.state('selectedItems')).length).toEqual(2);
```

Comment:
tiny nit: we've been trying to use `toBe` for primitive values
```suggestion
            expect(Object.keys(wrapper.state('selectedItems')).length).toBe(2);
```